tests/test_operator.py

```diff
@@ -30,6 +30,22 @@ def symbol(name, dimensions, value=0., mode='function'):
     return s.indexify() if mode == 'indexed' else s
 
 
+class TestAPI(object):
+
+    @classmethod
+    def setup_class(cls):
+        clear_cache()
+
+    def test_code_generation(self):
```

Comment:
>e.g. by searching for strings or inspecting ASTs through tools.pprint)
I am honeslty completly lost trying to got through `visitor` and figure out how it works.